profile-specific-challenges/frontend/components/idioms.jsx

```diff
@@ -392,34 +459,60 @@ export function UnoptimizableRenderingStructure(altRecords) {
 }
 
 // RenderHooks
-
-function useRenderHook(number) {
-  return <div>{number}</div>;
-}
-
 export function RenderHookComponent() {
   const [counter, setCounter] = useState();
-  const number = useRenderHook(counter);
 
   return (
     <div>
       <button onClick={() => setCounter((n) => n + 1)}>Click up</button>
-      {number}
+      <div>{counter}</div>
     </div>
   );
 }
 
 // Prop Drilling
```

Comment:
Changed this example to account for a more understandable scenario, when prop drilling instead of composition this would be something like: 

```jsx
export function ExcessivePropDrilling() {
  const counter = useState(0);
  const [count, setCount] = counter;

  return (
    <div>
      <p>Count: {count}</p>
      <button onClick={() => setCount((o) => o + 1)}>Increment</button>

      <Multiplications counter={counter} />
      <Divisions counter={counter} />
    </div>
  );
}

function MultBy5({ setCount }) {
  return (
    <button onClick={() => setCount((old) => old * 5)}>Multiply by 5</button>
  );
}

function Multiplications({ counter }) {
  const [count, setCount] = counter;
  return (
    <div>
      <p>Multiply count: {count}</p>
      <MultBy5 setCount={setCount} />
    </div>
  );
}

function DivideBy5({ setCount }) {
  return (
    <button onClick={() => setCount((old) => old / 5)}>Multiply by 5</button>
  );
}

function Divisions({ counter }) {
  const [count, setCount] = counter;
  return (
    <div>
      <p>Divide count: {count}</p>
      <DivideBy5 setCount={setCount} />
    </div>
  );
}
```